test/test_spectral_ops.py

```diff
@@ -1125,10 +1125,21 @@ def _test_istft_is_inverse_of_stft(stft_kwargs):
                 for i in range(num_trials):
                     original = torch.randn(*sizes, dtype=dtype, device=device)
                     stft = torch.stft(original, return_complex=True, **stft_kwargs)
-                    inversed = torch.istft(stft, length=original.size(1), **istft_kwargs)
-
+                    inversed = torch.istft(stft, length=original.size(-1), **istft_kwargs)
+                    n_frames = stft.size(-1)
+                    if stft_kwargs["center"] is True:
+                        len_expected = stft_kwargs["n_fft"] // 2 + stft_kwargs["hop_length"] * (n_frames - 1)
+                    else:
+                        len_expected = stft_kwargs["n_fft"] + stft_kwargs["hop_length"] * (n_frames - 1)
                     # trim the original for case when constructed signal is shorter than original
-                    original = original[..., :inversed.size(-1)]
+                    if original.size(-1) > len_expected:
```

Comment:
You need to add a high `hop_length` case that actually satisfies this condition. I also expect it'll need to be in a new test since the original signal will obviously not be fully preserved.

Reply:
Sure. I have added the tests in https://github.com/pytorch/pytorch/pull/63469/commits/716403d09866fcecfbd64d930693405d7c5a7320